File: plugins/gotemplate-kit/testdata/good/ok.go
package good

import (
	"html/template"
	"io"
	"strings"
)

// Documented intentional seam; keep allow on the smell line.
var legacy = template.HTML("<b>ok</b>") // gotemplate-rg-allow: fixture documents allow marker for intentional trusted-CMS HTML seam

func Render(w io.Writer, title string) error {
	funcs := template.FuncMap{
		"join": strings.Join,
	}
	t, err := template.New("ok").Funcs(funcs).Parse(`
		{{define "content"}}<h1>{{.}}</h1>{{end}}
		{{template "content" .}}
	`)
	if err != nil {
		return err
	}
	if err := t.ExecuteTemplate(w, "content", title); err != nil {
		return err
	}
	_ = legacy
	return nil
}

File: plugins/r-kit/testdata/good/ok.R
summarise_cars <- function(data = datasets::mtcars, verbose = TRUE) {
  if (isTRUE(verbose)) {
    message("n=", nrow(data))
  }
  mean(data$mpg, na.rm = TRUE)
}

# Documented intentional seam; keep allow on the smell line.
attach(list(x = 1)) # r-rg-allow: fixture documents allow marker for intentional attach seam
